java/code/src/com/suse/scc/model/SCCHwInfoJson.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2021 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.scc.model;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * This is a System Item send to SCC for registration.
+ */
+public class SCCHwInfoJson {
+
+    private int cpus;
+    private int sockets;
+    private String arch;
+    private String uuid;
+    private String hypervisor;
```

Comment:
I developed this along with those changes but in the end I thought this rework should have a separate pr (so it would not block the one actually adding the mem_total to the registration)